src/components/Header.tsx

```diff
@@ -1,20 +1,128 @@
 import React, { useEffect, useState, useContext } from 'react';
 import clsx from 'clsx';
 import Link from 'next/link';
-import { useRouter } from 'next/router';
-
-import { GithubIcon } from '../components/icons/GithubIcon';
+import Router, { useRouter } from 'next/router';
+import { Dialog } from '@headlessui/react';
 import { Logo } from '../components/Logo';
-import { MobileNavigation } from '../components/MobileNavigation';
 import { Search } from '../components/Search';
 import { ThemeSelector } from '../components/ThemeSelector';
 import { ThemeContext } from '../contexts/ThemeContext';
 
-interface HeaderProps {
-  navigation: any;
-}
+export function NavPopover({ display = 'lg:hidden', className, ...props }) {
+  const [isOpen, setIsOpen] = useState(false);
+
+  useEffect(() => {
+    if (!isOpen) return;
+    function handleRouteChange() {
+      setIsOpen(false);
+    }
+    Router.events.on('routeChangeComplete', handleRouteChange);
+    return () => {
+      Router.events.off('routeChangeComplete', handleRouteChange);
+    };
+  }, [isOpen]);
 
-export const Header: React.FC<HeaderProps> = ({ navigation }) => {
```

Comment:
We don't need to remove `HeaderProps`. In fact, we want to pass navigation items as props to this `Header` component. Please add this back.

Reply:
* add this

Reply:
Not required anymore.

Reply:
remove it